CMakeLists.txt

```diff
@@ -61,7 +61,7 @@ if(UNIX)
 endif()
 
 if(WIN32)
-    if((NOT MSVC) OR (MSVC_VERSION LESS 1900) OR (MSVC_VERSION GREATER 1919))
+    if((NOT MSVC) OR (MSVC_VERSION LESS 1930) OR (MSVC_VERSION GREATER 1934))
         message(SEND_ERROR "MSVC 2015 or 2017 is required")
```

Comment:
the message should say "MSVC 2022 is required"

Reply:
i recommend this diff:

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 37e657912..a9428a2d2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,8 +61,8 @@ if(UNIX)
 endif()

 if(WIN32)
-    if((NOT MSVC) OR (MSVC_VERSION LESS 1900) OR (MSVC_VERSION GREATER 1919))
-        message(SEND_ERROR "MSVC 2015 or 2017 is required")
+    if((NOT MSVC) OR (MSVC_VERSION LESS 1930) OR (MSVC_VERSION GREATER 1934))
+        message(SEND_ERROR "MSVC 2022 is required")
     endif()
 endif()

@@ -102,10 +102,10 @@ endif()

 # Automatically detect architecture based on Visual Studio generator
 if(MSVC AND NOT DEFINED DFHACK_BUILD_ARCH)
-    if(${CMAKE_GENERATOR} MATCHES "Win64")
-        set(DFHACK_BUILD_ARCH "64")
-    else()
+    if ( (${CMAKE_GENERATOR} MATCHES "Win32") OR (${CMAKE_GENERATOR} MATCHES "x86"))
         set(DFHACK_BUILD_ARCH "32")
+    else()
+        set(DFHACK_BUILD_ARCH "64")
     endif()
 else()
     set(DFHACK_BUILD_ARCH "64" CACHE STRING "Architecture to build ('32' or '64')")
```

Reply:
done